lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -110,7 +118,12 @@ def create
             args.push('--type', @resource[:type])
         end
 
-        args << @resource[:volume_group]
+        if @resource[:thin]
+            args.push('--thin')
+            args << @resource[:volume_group] + "/" + @resource[:name]
```

Comment:
as you quoted from the man page. `--thin` seems to have multiple outcomes. but there is `--thinpool` too. for me it seems like `--thinpool` was introduced later to be specific what happens.